lua/nvim-treesitter/node_movement.lua

```diff
@@ -1,88 +1,76 @@
 local api = vim.api
-local parsers = require'nvim-treesitter.parsers'
-local utils = require'nvim-treesitter.utils'
-local M = {}
 
+local configs = require'nvim-treesitter.configs'
+local state = require'nvim-treesitter.state'
+local ts_utils = require'nvim-treesitter.ts_utils'
+
+local M = {}
 
-M.NodeMovementKind = {
+local NodeMovementKind = {
   up = 'up',
   down = 'down',
   left = 'left',
   right = 'right',
 }
 
-M.current_node = {}
-
-local function node_start_to_vim(node)
-  if not node then return end
-
-  local row, col = node:start()
-  local exec_command = string.format('call cursor(%d, %d)', row+1, col+1)
-  api.nvim_exec(exec_command, false)
-end
+local get_node_fn = {
+  [NodeMovementKind.up] = function(node)
+    return node:parent()
+  end,
+  [NodeMovementKind.down] = function(node)
+    if node:named_child_count() > 0 then
+      return node:named_child(0)
+    else
+      local next_node = ts_utils.get_next_node(node)
+      if next_node and next_node:named_child_count() > 0 then
+        return next_node:named_child(0)
+      end
+    end
+  end,
+  [NodeMovementKind.left] = function(node)
+    return ts_utils.get_previous_node(node, true, true)
+  end,
+  [NodeMovementKind.right] = function(node)
+    return ts_utils.get_next_node(node, true, true)
+  end,
+}
 
 M.do_node_movement = function(kind)
-  local buf, line, col = unpack(vim.fn.getpos("."))
+  local buf = api.nvim_get_current_buf()
```

Comment:
Finally, this is fixed :yum: ! My bad that I didn't realize this in the beginning.

Reply:
actually the node_movement is not working properly, i'm currently rewriting it